charts/netbox/templates/deployment.yaml

```diff
@@ -109,8 +109,26 @@ spec:
         - name: http
           containerPort: 8080
           protocol: TCP
-        {{- if .Values.readinessProbe.enabled }}
-        readinessProbe:
+        - name: nginx-status
+          containerPort: 8081
+          protocol: TCP
+        {{- if .Values.customLivenessProbe }}
+        livenessProbe: {{- include "common.tplvalues.render" (dict "value" .Values.customLivenessProbe "context" $) | nindent 10 }}
+        {{- else if .Values.livenessProbe.enabled }}
```

Comment:
```
Error: UPGRADE FAILED: template: netbox/templates/deployment.yaml:117:27: executing "netbox/templates/deployment.yaml" at <.Values.livenessProbe.enabled>: nil pointer evaluating interface {}.enabled
```

I guess to handle upgrades should we make this something like:
```
{{- if and (hasKey .Values.readinessProbe "enabled") .Values.readinessProbe.enabled }}
```

I'm honestly surprised there isn't a bitnami common template for "truthiness" like this. Feels like it would be a common issue.

Reply:
@RangerRick I believe that will generate the same error: `.Values.readinessProbe` is `nil`.
As mentioned in the PR description, that would require a major version bump.
Indeed, `chart-testing` expects minor and patch versions to be values-compatible with each others, under the same major version. It is testing if you can use the `values.yaml` from the previous version with the new chart version, and vice versa. 

Two ways to make it happy:
* Bump the chart to a new major version, it is then clear that there is a breaking change.
* Add additional value conditions in the templates, with the risk of overcomplicating.

Hope it makes sense 😅

Reply:
Since we've been treating this as "release-candidate" ready, and I know there's definitely folks using it in the real world, it would be nice to add some stuff to the templates, despite being annoyingly overcomplicated. I doubt this will be the last time we want to avoid breaking things just because we're adding a small feature improvement.

I'm OK with breaking it this time around if it's going to be too much trouble, but it might be nice to come up with a way to handle stuff like this gracefully on upgrade, when there's nothing technically _wrong_ with someone's existing values. It's a pain point for the user that doesn't _need_ to exist if we can come up with a nice way to handle it in the template without a bunch of boilerplate.